swagger.yaml

```diff
@@ -1,42 +1,292 @@
+        404:
+          $ref: '#/responses/NotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    put:
+      tags:
+      - "image"
+      summary: "Update an image metadata entry"
+      description: "Updates an image metadata entry whose id matches the id provided as the last element of the path"
+      parameters:
+      - $ref: '#/parameters/image_id'
+      - $ref: '#/parameters/image'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "A json with the requested image metadata"
+          schema:
+            $ref: '#/definitions/Image'
+        400:
+          description: |
+            Invalid request, reasons can be one of the following:
+              * image id was incorrect
+              * collection id was incorrect
+              * malformed body
+              * provided image had an invalid parameter
+        401:
+          description: "Unauthorised to update image metadata"
+        404:
+          $ref: '#/responses/NotFound'
+        409:
+          description: "Cannot update an image for the provided collection ID because it is already published"
         500:
           $ref: '#/responses/InternalError'
 
 responses:
+
   InternalError:
     description: "Failed to process the request due to an internal error"
+    
+  NotFound:
+    description: "Requested item cannot be found"
+    
+  InvalidRequestError:
+    description: "Failed to process the request due to invalid request"
 
 definitions:
-  HelloResponse:
+
+  Images:
+    description: "A list of images"
     type: object
     properties:
-      message:
+      count:
+        description: "The number of images returned"
+        readOnly: true
+        type: integer
+        example: 1
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Image'
+      limit:
+        description: "The number of images requested"
+        type: integer
+      offset:
+        description: "The first row of images to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of images"
+        readOnly: true
+        type: integer
+        example: 1
+
+  Image:
+    type: object
+    description: "Image metadata model"
+    properties:
+      id:
+        type: string
+        description: "Image metadata unique identifier"
+        example: "042e216a-7822-4fa0-a3d6-e3f5248ffc35"
+      collection_id:
+        type: string
+        description: "Collection unique identifier corresponding to this image"
+        example: "5557dcd9-bf58-4a67-94f7-2343569834cc"
+      state:
+        type: string
+        description: |
+          The state of the image, can only be one of the following:
+          * created
+          * published
```

Comment:
Yes, and I also double-checked the diagrams, and I created a new `POST /images/{id}/publish` endpoint, which returns when we do a publish request, so I think it would also be worth to create a `publishing` (then the static file publisher will update it to `published` once complete)